pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -5864,6 +5864,70 @@ func TestChangefeedOutdatedCursor(t *testing.T) {
 	cdcTestWithSystem(t, testFn, feedTestNoTenants)
 }
 
+// TestChangefeedCursorWarning ensures that we show a warning if
+// any of the tables we're creating a changefeed is past
+// the warning threshold.
+func TestChangefeedCursorAgeWarning(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	s, stopServer := makeServer(t)
+	defer stopServer()
+
+	sqlDB := sqlutils.MakeSQLRunner(s.DB)
+	sqlDB.Exec(t, `CREATE TABLE f (a INT PRIMARY KEY)`)
+	sqlDB.Exec(t, `INSERT INTO f VALUES (1)`)
+	timeNow := strings.Split(s.Server.Clock().Now().AsOfSystemTime(), ".")[0]
+	noWarning := "(no notice)"
+
+	// Using default warning threshold age of 5 hours
+	t.Run("default threshold", func(t *testing.T) {
+		expectNotice(t, s.Server,
+			fmt.Sprintf(
+				`CREATE CHANGEFEED FOR TABLE d.f INTO 'null://' with cursor = '%s', initial_scan='only'`,
+				timeNow), noWarning)
+
+		expectNotice(t, s.Server,
+			fmt.Sprintf(
+				`CREATE CHANGEFEED FOR TABLE d.f INTO 'null://' with cursor = '%s', initial_scan='yes'`,
+				timeNow), noWarning)
+
+		expectNotice(t, s.Server,
+			fmt.Sprintf(
+				`CREATE CHANGEFEED FOR TABLE d.f INTO 'null://' with cursor = '%s', initial_scan='no'`,
+				timeNow), noWarning)
+
+	})
+
+	overrideCursorAge := int64(6 * time.Hour)
+	knobs := s.TestingKnobs.
+		DistSQL.(*execinfra.TestingKnobs).
+		Changefeed.(*TestingKnobs)
+	knobs.OverrideCursorAge = func() int64 {
```

Comment:
Looks like there's a data race on Bazel Extended CI. I think moving this block into the `t.Run("lowered threshold", ...` should fix it.